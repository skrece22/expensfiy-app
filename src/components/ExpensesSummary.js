import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary= ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedTotal}</h1>
        </div>
    );
}

const mapStateToProps = (state) => {
    const visibileExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expenseCount: visibileExpenses.length,
        expensesTotal: selectExpensesTotal(visibileExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);