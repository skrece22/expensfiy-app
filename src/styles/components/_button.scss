.button{
    background: $blue;
    border: 0;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link{
    background: none;
    border: none;
    color: white;
}

.button--secondary{
    background: #888888;
}