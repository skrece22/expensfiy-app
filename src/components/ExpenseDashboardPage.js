import React from 'react';
import ExpenseList from './ExpenseList';
import ExpeneListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = ()=>(
    <div>
        <ExpensesSummary />
        <ExpeneListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;