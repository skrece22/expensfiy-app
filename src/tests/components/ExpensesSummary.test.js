import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test("should render expenses with one expense", ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={6045}/>);
    expect(wrapper).toMatchSnapshot();
});

test("should render expenses with mulitiple expenses", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={1246025} />);
    expect(wrapper).toMatchSnapshot();
});