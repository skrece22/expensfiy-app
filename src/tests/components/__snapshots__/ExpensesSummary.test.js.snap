// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses with mulitiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $12,460.25
  </h1>
</div>
`;

exports[`should render expenses with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $60.45
  </h1>
</div>
`;
